ml-agents-envs/mlagents/envs/sampler_class.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+from functools import *
+from abc import ABC, abstractmethod
+
+from .exception import SamplerException
+
+class SamplerException(Exception):
+    pass
+
+class Sampler(ABC): 
+
+    @abstractmethod
+    def sample_parameter(self, *args, **kwargs):
+        pass
+
+
+class UniformSampler(Sampler):
+    # kwargs acts as a sink for extra unneeded args
+    def __init__(self, min_value, max_value, **kwargs):
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def sample_parameter(self):
+        return np.random.uniform(self.min_value, self.max_value)
+    
+
+class MultiRangeUniformSampler(Sampler):
+    def __init__(self, intervals, **kwargs):
+        self.intervals = intervals
+        # Measure the length of the intervals
+        self.interval_lengths = list(map(lambda x: abs(x[1] - x[0]), self.intervals))
```

Comment:
I find list comprehensions more readable than maps and reduces (and they were also removed from python3). I think this works:
```
self.interval_lengths = [abs(x[1] - x[0]) for x in self.intervals]
self.cum_interval_length = sum(self.interval_lengths)
self.interval_weights = [x / self.cum_interval_length for x in self.interval_lengths]
```
and I also think you can keep interval_lengths and cum_interval_length as local variables.

Reply:
Converted them to local variables